### DISCUSSION TITLE
How can I fix error #418 from console?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#64470] [Help] [answered]
### Summary

Error is this:
![image](https://github.com/vercel/next.js/assets/93617433/00250b55-fadb-4d5e-a0d0-a94553664501)

^^^ This seems to be a random error that I am getting on some (not all) of the blog pages, seems weird, because why showing only on some pages but not all? when all the blog pages have the same structure?

I was thinking it had to do with some of the added html, but other pages also have added html..

How can I detect where is the error coming from and how to fix it???

One of the pages showing this error is this one:
https://github.com/Elindo586/nextprojectb/blob/main/pages/blog/english/v10-v20-vane-pump-catalog.js

```
import React from "react";
// import { useEffect } from "react";
import Blogger2 from "../../../components/blog-post-english";

import db from "../../../utils/blogs-front/english/blog-english.json";

export async function getStaticProps() {
  return {
    props: { db },
  };
}


const PostEnglish46 = ({ db }) => {
  // This is another way to do it if I dont do filter below
  // function post(info) {
  //   return (
  //     info.id === "5",
  //     info.item,
  //     info.metaTitle,
  //     info.metaDescription,
  //     info.metaKeywords,
  //     info.ogTitle,
  //     info.ogDescription,
  //     info.ogURL,
  //     info.ogImage,
  //     info.twitterTitle,
  //     info.twitterDescription,
  //     info.twitterImage,
  //     info.title,
  //     info.body,
  //     info.description,
  //     info.previous,
  //     info.next
  //   );
  // }

  // const item = db.find(post);

  return (
    <div>
      {db
        .filter((item) => item.id === "46")
        .map((item) => {
          return (
            <Blogger2
              key={item.id}
              metaTitle={item.metaTitle}
              metaDescription={item.metaDescription}
              metaKeywords={item.metaKeywords}
              ogTitle={item.ogTitle}
              ogDescription={item.ogDescription}
              ogURL={item.ogURL}
              ogImage={item.ogImage}
              twitterTitle={item.twitterTitle}
              twitterDescription={item.twitterDescription}
              twitterImage={item.twitterImage}
              title={item.title}
              body={item.body}
              iTitle={item.iTitle}
              description={
                <div
                  dangerouslySetInnerHTML={{ __html: item.description }}
                ></div>
              }

/// other pages showing this error don't have the dangerouslySetInnerHTML, instead I just add a full article html in this section..... other pages show no errors with the dangerouslySetInnerHTML..
              previous={item.previous}
              next={item.next}
            />
          );
        })}
    </div>
  );
};

export default PostEnglish46;



```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This error should also be visible in development mode, there you will get more information about the issue.

From your description, I'd guess that the issue is most likely invalid HTML. When the browser encounters invalid HTML, it will restructure it the HTML into a valid shape, notably, nesting `p` tags such as:

```html
<p>
   <p>Hello</p>
</p>
```

Will be restructured to:

```html
<p> </p>
<p>Hello</p>
<p></p>
```

And like that, there is a few other cases, where the HTML will be restructured, and end up causing a hydration error.

> Note that this is for HTML sent to the browser. Programmatically, you can append a `p` tag as child to another `p` tag, which is why this error usually pop up when using an SSR framework.

Or as the link given by the React warning says, anything that causes divergent HTML, for instance, dates, or rendering threes based of window as your render.

In your particular case, I think it is just a case of a `div` nested within a `p` tag, which is also invalid HTML. 

In, https://github.com/Elindo586/nextprojectb/blob/main/components/blog-post-english.js#L108C1-L111C17, you have:

```tsx
          <div className="">
            {" "}
            <p> {props.description}</p>{" "}
          </div>
```

And in the code you are showing use, `props.description`, is a `div`, that'll throw for sure.

- https://caninclude.glitch.me/caninclude?child=div&parent=p

Once again, your server renders something like:

```html
<p>
  <div>hi</div>
</p>
```

And then the browser, because as it is, it is invalid HTML, restructures it to:

```html
 <p> </p>
 <div>hi</div>
 <p></p>
```

So React sees a different HTML structure than it expected, and boom 418 is raised.

Remember that one can't nest `p` tags either, so maybe change to something like:

```tsx
          <div className="">
            {" "}
            <>{props.description}</>{" "}
          </div>
```

And maybe remove all of those white spaces, if they are not useful. Then let the description define its wrapper, be it a `p`, or a `div` tag. You'll need to think through your use cases here.

**Reply by Elindo586** [reply]
Well, that fixed it.

`<p>` tags have been a problem before, and I didn't know I had that one....  

I ran it on dev and got more information... would have taken me a while to find it (or never).... don't know how you found it.
